### PR TITLE
Feature/5 rent book

### PR DESCRIPTION
**Pull request by Zokster9**
Implement rent and return functionality

### CODE REVIEW

File: library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx [hunk 1]
```diff
@@ -52,6 +58,22 @@ const BookDetailsPage = () => {
       notifyBookCantBeDeleted()
     }
   }
+  const handleRentBook = () => {
+    if (!bookId) return
+    if (book?.Available !== 0) {
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx:63]: What if it's less then 0?

File: library-valcon-internship/src/services/RentService.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import BookRentHistoryResponse from '../models/responses/BookRentHistoryResponse'
+import { baseUrl } from './AxiosConfiguration'
+
+export const rentBook = async (bookId: string) => {
+  return axios.post(baseUrl + `api/Rental/rent/${bookId}`)
+}
+
+export const getBookRentHistory = async (bookId: string) => {
+  return axios.get<BookRentHistoryResponse[]>(baseUrl + `api/Rental/book-history/${bookId}`)
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/services/RentService.ts:11]: Don't combine template literal with string operators

File: library-valcon-internship/src/services/RentService.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import axios from 'axios'
+
+import BookRentHistoryResponse from '../models/responses/BookRentHistoryResponse'
+import { baseUrl } from './AxiosConfiguration'
+
+export const rentBook = async (bookId: string) => {
+  return axios.post(baseUrl + `api/Rental/rent/${bookId}`)
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/services/RentService.ts:7]: same as bellow

File: library-valcon-internship/src/services/RentService.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export const getBookRentHistory = async (bookId: string) => {
+  return axios.get<BookRentHistoryResponse[]>(baseUrl + `api/Rental/book-history/${bookId}`)
+}
+
+export const returnBook = async (bookId: string, userId: number) => {
+  return axios.post(baseUrl + `api/Rental/return/${bookId}/${userId}`)
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/services/RentService.ts:15]: same

File: library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx [hunk 5]
```diff
@@ -107,35 +129,59 @@ const BookDetailsPage = () => {
...
+            }
           </div>
         </div>
       </div>
       <DeleteBookDialog dialogRef={dialogRef} handleConfirm={removeBook} />
       {showModal &&
         <BookFormWrapper retrieveBook={retrieveBook} setRetrieveBook={setRetrieveBook} book={book} closeModal={handleCloseModal} />
       }
+      {
+        token.Role !== 'User' &&
```
Comment (SaleBojke) [inline library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx:177]: Ja bih radije proveravao da li je admin ili librarian nego da li nije user.

File: library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx [hunk 6]
```diff
@@ -107,35 +129,59 @@ const BookDetailsPage = () => {
...
+              <button
+                type='button'
+                className='book-details-btn rent'
+                onClick={handleRentBook}
+              >
+                Rent the book
+              </button>
+            }
+            {
+              token.Role !== 'User' &&
```
Comment (SaleBojke) [inline library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx:144]: isto

File: library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx [hunk 7]
```diff
@@ -107,35 +129,59 @@ const BookDetailsPage = () => {
...
-            <button
-              type='button'
-              className='book-details-btn delete'
-              onClick={handleDeleteBook}
-            >
-              Delete book
-            </button>
+          <div className='book-details-actions-field'>
+            {
+              token.Role !== 'Librarian' &&
```
Comment (SaleBojke) [inline library-valcon-internship/src/pages/BookDetailsPage/BookDetailsPage.tsx:134]: obrnuto

**Review by SaleBojke — APPROVED** [review]

**Pull request merged**